frontend/projects/upgrade/src/app/features/dashboard/segments/pages/segment-details-page/segment-details-page.component.ts

```diff
@@ -1,12 +1,71 @@
-import { Component } from '@angular/core';
+// segment-details-page.component.ts
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { CommonPageComponent } from '../../../../../shared-standalone-component-lib/components';
 import { SegmentDetailsPageHeaderComponent } from './segment-details-page-header/segment-details-page-header.component';
 import { SegmentDetailsPageContentComponent } from './segment-details-page-content/segment-details-page-content.component';
+import { SegmentsModule } from '../../../segments-legacy/segments.module';
+import { SegmentsService } from '../../../../../core/segments/segments.service';
+import { ActivatedRoute } from '@angular/router';
+import { Segment } from '../../../../../core/segments/store/segments.model';
+import { SEGMENT_TYPE } from 'upgrade_types';
+import { Observable, Subscription } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { CommonModule } from '@angular/common';
 
 @Component({
   selector: 'app-segment-details-page',
   templateUrl: './segment-details-page.component.html',
   styleUrl: './segment-details-page.component.scss',
-  imports: [CommonPageComponent, SegmentDetailsPageHeaderComponent, SegmentDetailsPageContentComponent],
+  imports: [
+    CommonModule,
+    CommonPageComponent,
+    SegmentDetailsPageHeaderComponent,
+    SegmentDetailsPageContentComponent,
+    SegmentsModule,
+  ],
+  standalone: true,
 })
-export class SegmentDetailsPageComponent {}
+export class SegmentDetailsPageComponent implements OnInit, OnDestroy {
+  segment$: Observable<Segment>;
+  segmentIdSub: Subscription;
+  shouldUseLegacyView$: Observable<boolean>;
+
+  constructor(private segmentsService: SegmentsService, private route: ActivatedRoute) {}
+
+  ngOnInit() {
+    this.segmentIdSub = this.route.paramMap.subscribe((params) => {
+      const segmentId = params.get('segmentId');
+      if (segmentId) {
+        this.segmentsService.fetchSegmentById(segmentId);
+      }
+    });
+
+    this.segment$ = this.segmentsService.selectedSegment$.pipe(filter((segment) => !!segment));
+
+    // Determine if we should use the legacy view based on segment structure
+    this.shouldUseLegacyView$ = this.segment$.pipe(map((segment) => this.shouldUseLegacyUI(segment)));
+  }
+
+  // Logic to determine if we should use the legacy UI
+  private shouldUseLegacyUI(segment: Segment): boolean {
```

Comment:
nice, i'd suggest making this a selector, or at least put into service. It would work essentially the same, but ngrx selectors are built for this exact kind of purpose, it's a data-layer job not a component job.